### ISSUE TITLE
20.31.5 broken

### ISSUE THREAD

**Issue by zachvida** [terraform-aws-modules/terraform-aws-eks#3254]
## Description
Terraform v1.6.4
on linux_amd64
Initializing plugins and modules...
module.eks.aws_eks_cluster.this[0]: Modifying... [id=cluster-dev]
╷
│ Error: updating EKS Cluster (cluster-dev) compute config: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: XXXXX, InvalidParameterException: The type for cluster update was not provided.
│ 
│   with module.eks.aws_eks_cluster.this[0],
│   on .terraform/modules/eks/main.tf line 35, in resource "aws_eks_cluster" "this":
│   35: resource "aws_eks_cluster" "this" {
│ 

I believe this is related to the fact that govcloud does not support  the automode features yet.

Operation failed: failed running terraform apply (exit 1)
- [x ] ✋ I have searched the open/closed issues and my issue is not listed.

Cloudtail logs snippet:
errorCode: InvalidParameterException
eventName: UpdateClusterConfig
 eventSource: eks.amazonaws.com
requestParameters": {"clientRequestToken": "33333", "storageConfig": {"blockStorage": {"enabled": false}}, "name": "cluster-dev", "kubernetesNetworkConfig": {"elasticLoadBalancing": {"enabled": false}, "ipFamily": "ipv4", "serviceIpv4Cidr": "172.20.0.0/16"}}, "responseElements": {"message": "The type for cluster update was not provided."}

## Versions

- Module version [Required]: 20.31.5 reversion to 20.31.1 works

- Terraform version:
1.6.4
- Provider version(s):
5.82.0

**Comment by waitstime** [comment]
Looks like it was broken by https://github.com/terraform-aws-modules/terraform-aws-eks/pull/3253

**Comment by davekonopka** [comment]
We're hitting an error on regular AWS cloud related to the storage change linked above for existing clusters managed with the module:

```
│ Error: updating EKS Cluster (...) compute config: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: ..., InvalidParameterException: The type for cluster update was not provided.
```

It wants to add the `storage_config` block to the cluster:

```
      + storage_config {
          + block_storage {
              + enabled = false
            }
        }
```

**Comment by kian** [comment]
Ran into this as well (AWS commercial; it's not GovCloud-specific).  The issue appears to be that `compute_config` must be [disabled when not using Auto Mode](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#eks-cluster-with-eks-auto-mode).

20.31.5 aims to fix this, but I think `compute_config` with `enabled = false` [does not get created](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/3253/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR56) when `cluster_compute_config` is not set by module users.

I worked around it by being explicit when invoking the module:

```
module "eks" {
  ...
  cluster_compute_config = {
    enabled = false
  }
}
```

**Comment by zachvida** [comment]
updated the issue title

**Comment by bdandoy** [comment]
We are having the same issue @bryantbiggs can you help?

**Comment by bryantbiggs** [comment]
oy, this API shape across 3 attribute is quite problematic. I think the only course is to revert the prior change

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.31.6 :tada:

**Comment by moore-nathan** [comment]
I'll ask here first before making a new issue, but does the revert mean that the reverse of this is broken?
I get a `compute config: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: xxx, InvalidParameterException: The type for cluster update was not provided.` when trying to explicitly set it to false. (This is for an already existing cluster)
```yaml
cluster_compute_config =  {
    enabled       =  false
    node_pools = ["general-purpose", "system"]
}
```
